File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/view/dialogs/NetworkServerErrorDialog.java
package it.dani.selfhomeapp.middle.view.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.function.Supplier;

import it.dani.selfhomeapp.R;

public class NetworkServerErrorDialog extends DialogFragment {

    private Supplier<Void> positiveSupplier;
    private Supplier<Void> negativeSupplier;

    public NetworkServerErrorDialog()
    {
        super();
        this.positiveSupplier = null;
        this.negativeSupplier = null;
    }

    public NetworkServerErrorDialog(@NonNull Supplier<Void> positiveSupplier, @NonNull Supplier<Void> negativeSupplier)
    {
        super();
        this.positiveSupplier = positiveSupplier;
        this.negativeSupplier = negativeSupplier;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstance)
    {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        builder.setTitle(R.string.titleHandleNoWifiNetworkResponseErr);
        builder.setMessage(R.string.noWifiNetworkResponseErr);
        builder.setPositiveButton(R.string.positiveHandleNoWifiNetworkResponseErr,(x,y) -> {
            if(this.positiveSupplier != null) this.positiveSupplier.get();
        });
        builder.setNegativeButton(R.string.negativeHandleNoWifiNetworkResponseErr,(x,y) -> {
            if(this.negativeSupplier != null) this.negativeSupplier.get();
        });

        return builder.create();
    }
}

File: SelfHomeApp/app/src/main/java/it/dani/selfhomeapp/middle/view/dialogs/NetworkServerErrorDialog.kt
package it.dani.selfhomeapp.middle.view.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import it.dani.selfhomeapp.R

class NetworkServerErrorDialog(private val positiveSupplier : () -> Unit, private val negativeSupplier : () -> Unit) : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(this.activity).apply {
            setTitle(R.string.titleHandleNoWifiNetworkResponseErr)
            setMessage(R.string.noWifiNetworkResponseErr)
            setPositiveButton(R.string.positiveHandleNoWifiNetworkResponseErr) { _, _ ->
                this@NetworkServerErrorDialog.positiveSupplier()
            }
            setNegativeButton(R.string.negativeHandleNoWifiNetworkResponseErr) {_,_ ->
                this@NetworkServerErrorDialog.negativeSupplier()
            }
        }.create()
    }
}